tests/parameter_handler/parameter_handler_6_spaces.cc

```diff
@@ -0,0 +1,101 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2003 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+
+// This is a variation of the parameter_handler_6 test. At some point,
+// we made an inadvertant change that led to no longer correctly
+// indenting the parameters listed in a section when outputting the
+// current state of a ParameterHandler object. numdiff didn't catch it
+// because it ignores whitespace except to separate tokens. This test
+// therefore outputs everything into a stringstream, replaces every
+// space by an underscore, and outputs that. Changes in indentation
+// will then show up as changes in output in a way that can't be
+// ignored by numdiff.
```

Comment:
Thanks for checking this out - this explains why we didn't catch the bug earlier.